src/main/app/src/components/suodattimet/hakutulosSuodattimet/SuodatinValinnat.tsx

```diff
@@ -97,18 +102,12 @@ export const SuodatinValinnat = () => {
   const getHandleDelete = (item: FilterValue) => () => {
     const changes = getFilterStateChanges(selectedFiltersWithAlakoodit)(item);
 
-    let changesWithBooleanValues;
-    if (item.filterId === FILTER_TYPES.HAKUKAYNNISSA) {
-      changesWithBooleanValues = { ...changes, hakukaynnissa: !item.checked };
-    } else {
-      changesWithBooleanValues = _.omit(changes, 'hakukaynnissa');
-    }
-
-    if (item.filterId === FILTER_TYPES.JOTPA) {
-      changesWithBooleanValues = { ...changesWithBooleanValues, jotpa: !item.checked };
-    } else {
-      changesWithBooleanValues = _.omit(changesWithBooleanValues, 'jotpa');
-    }
+    const changesWithBooleanValues = filtersWithBooleanValues.includes(item.filterId)
+      ? _.omit(
+          { ...changes, [item.filterId]: !item.checked },
+          ...filtersWithBooleanValues.filter((f) => f !== item.filterId)
```

Comment:
Tän vois tehdä hieman luettavammin näin:
```suggestion
          _.without(filtersWithBooleanValues, item.filterId)
```

Reply:
Ja tosiaan tuo toinen parametri `omit`:lle voi olla taulukko, eli ei tarvi spreadia (`...`)